linear_operator/operators/block_diag_linear_operator.py

```diff
@@ -56,7 +57,7 @@ def __init__(self, base_linear_op, block_dim=-3):
             )
 
     @property
-    def num_blocks(self):
+    def num_blocks(self) -> int:
         return self.base_linear_op.size(-3)
 
     def _add_batch_dim(self, other):
```

Comment:
```suggestion
    def _add_batch_dim(self, other: Float[torch.Tensor, "... N C"]) -> Float[torch.Tensor, "... C"]:
```

Reply:
Not sure if this can be expressed properly with `jaxtyping`?

Reply:
Maybe it can. Talking with the jaxtyping author, functions and operators are supported over dimensions. The function definition is reproduced below:
```
    def _add_batch_dim(self, other):
        *batch_shape, num_rows, num_cols = other.shape
        batch_shape = list(batch_shape)

        batch_shape.append(self.num_blocks)
        other = other.view(*batch_shape, num_rows // self.num_blocks, num_cols)
        return other
```

Which suggests the following signature using the div operator. (This passes runtime tests):

```
def _add_batch_dim(self: Float[LinearOperator, "*batch1 M P"], other: Float[torch.Tensor, "*batch2 N C"]) \
            -> Float[torch.Tensor, "batch2 N//batch1 C"]:
```

@dannyfriar What do you think?

Reply:
Sorry for the late reply I missed this. It looks good to me, clearer than what was there before